sdk/tests/deqp/modules/shared/glsDrawTests.js

```diff
@@ -1524,228 +1475,56 @@ goog.scope(function() {
-
-                //Advance one vertex
-                writePtr = writePtr.subarray(componentCount * componentSize);
-                continue;
-            }
-
-            //Primitive cases
-            switch (primitive) {
-                case glsDrawTests.DrawTestSpec.Primitive.LINES:
-                case glsDrawTests.DrawTestSpec.Primitive.LINE_STRIP:
-                case glsDrawTests.DrawTestSpec.Primitive.LINE_LOOP: {
-                    quadVertices = [generateVertex(components, vertexParameters)];
-                    break;
-                }
-                case glsDrawTests.DrawTestSpec.Primitive.TRIANGLES: {
-                    vertex1 = generateVertex(components, vertexParameters);
-
-                    parallelaxis = rnd.getBool() ? 0 : 1;
-                    oppositeaxis = parallelaxis ? 0 : 1;
-
-                    vertex2 = generateVertex(vertex1, vertexParameters);
-                    vertex2[parallelaxis] = vertex1[parallelaxis];
-
-                    vertex3 = generateVertex(vertex2, vertexParameters);
-                    vertex3[oppositeaxis] = vertex1[oppositeaxis];
-
-                    vertex4 = generateVertex(vertex3, vertexParameters);
-                    vertex4[parallelaxis] = vertex3[parallelaxis];
-                    vertex4[oppositeaxis] = vertex2[oppositeaxis];
-
-                    quadVertices = [vertex1, vertex2, vertex3, vertex3, vertex2, vertex4];
-                    break;
-                }
-                case glsDrawTests.DrawTestSpec.Primitive.TRIANGLE_STRIP: {
-                    if (vertexNdx < first + 4) {
-                        vertex1 = generateVertex(components, vertexParameters);
-
-                        parallelaxis = rnd.getBool() ? 0 : 1;
-                        oppositeaxis = parallelaxis ? 0 : 1;
-
-                        vertex2 = generateVertex(vertex1, vertexParameters);
-                        vertex2[parallelaxis] = vertex1[parallelaxis];
-
-                        vertex3 = generateVertex(vertex2, vertexParameters);
-                        vertex3[oppositeaxis] = vertex1[oppositeaxis];
-
-                        direction = packed ?
-                            (vertex3[parallelaxis] > vertex1[parallelaxis] ? 1 : -1) :
-                            (vertex3[parallelaxis].greaterThan(vertex1[parallelaxis]) ? 1 : -1);
-
-                        vertex4 = generateVertex(vertex3, vertexParameters);
-                        vertex4[oppositeaxis] = vertex2[oppositeaxis];
-                        vertex4[parallelaxis] = vertex3[parallelaxis];
-
-                        quadVertices = [vertex1, vertex2, vertex3, vertex4];
+            var components = [];
+
+            for (var componentNdx = 0; componentNdx < componentCount; componentNdx++) {
+                var getRandomComponent = function() {
+                    // For packed formats we can't use GLValue
+                    if (packed) {
```

Comment:
The C++ version use templating on the vertex attribute type, while here we use GLValue that doesn't handle the (U)INT_2_10_10_10_REV formats so we special case these formats with the packed variable like the previous code was doing in various places.